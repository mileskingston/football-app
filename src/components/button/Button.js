import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Button = (props) => {
  return (
    <button
      className={classNames('btn', { 'btn--disabled' : props.disabled, [props.classes]: props.addClass })}
      value={props.value}
      type={props.type}
      disabled={props.disabled}
      onClick={props.onClick}
    >
      {props.children}
    </button>
  );
}

Button.propTypes = {
  children: PropTypes.node,
  value: PropTypes.string,
  type: PropTypes.string.isRequired,
  disabled: PropTypes.bool.isRequired,
  onClick: PropTypes.func,
  addClass: PropTypes.string
};

Button.defaultProps = {
  type: 'button',
  disabled: false,
  onClick: () => {},
}

export default Button;
