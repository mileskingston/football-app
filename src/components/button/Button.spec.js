import React, { Component } from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
  let props = {

  };

  describe('generic props', () => {
    it('should match snapshot', () => {
      expect(
        shallow(
          <Button>Submit</Button>
        )
      ).toMatchSnapshot();
    });
  });
  
  describe('generic props', () => {
    it('should match snapshot', () => {
      expect(
        shallow(
          <Button>Submit</Button>
        )
      ).toMatchSnapshot();
    });
  });
});
