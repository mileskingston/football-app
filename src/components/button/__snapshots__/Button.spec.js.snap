// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button props should match snapshot 1`] = `
<button
  className="btn"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Hello
</button>
`;
